pkg/bicep/biceptoschema.go

```diff
@@ -0,0 +1,231 @@
+package bicep
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"slices"
+
+	bp "github.com/Checkmarx/kics/v2/pkg/parser/bicep"
+	"github.com/massdriver-cloud/airlock/pkg/schema"
+
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+)
+
+type bicepParam struct {
+	TypeString    string             `json:"type"`
+	DefaultValue  interface{}        `json:"defaultValue"`
+	AllowedValues []interface{}      `json:"allowedValues"`
+	Metadata      bicepParamMetadata `json:"metadata"`
+	MinLength     *uint64            `json:"minLength"`
+	MinValue      *uint64            `json:"minValue"`
+	MaxLength     *uint64            `json:"maxLength"`
+	MaxValue      *uint64            `json:"maxValue"`
+}
+
+type bicepParamMetadata struct {
+	Description string                 `json:"description"`
+	Metadata    map[string]interface{} `json:"metadata"`
+}
+
+func BicepToSchema(valuesPath string) (string, error) {
+	// using the github.com/Checkmarx/kics parser since he already did the heavy lifting to parse a bicep template
+	parser := bp.Parser{}
+
+	params := new(schema.Schema)
+	params.Type = "object"
+	params.Properties = orderedmap.New[string, *schema.Schema]()
```

Comment:
assuming this is just using the above library to create an ordered map as opposed to unordered?

Reply:
Yeah, we could just use a `map[string]interface{}` but that is inherently unordered. JSON is technically unordered, so it doesn't matter, but for organization and presentation it makes it a bit nicer.